### ISSUE TITLE
Subsequent syncing of messages

### ISSUE THREAD

**Issue by bugsug** [42wim/matterbridge#825]
**Is your feature request related to a problem? Please describe.**
When the bridge goes down and is offline for a while or is only online for a few hours a day the messages which are sent during offtime in the bridged rooms are not bridged.

I run a Matrix <-> WhatsApp bridge on my desktop computer. Due to the low message rate in the bridged rooms it would be sufficent to sync the messages only a few times a day. I also do not want my computer to run the whole day when there's no necessarity.

**Describe the solution you'd like**
Every time Matterbridge starts, it checks for the last messages in the bridged rooms and syncs the messages which are not bridged yet. It would also be possible to save the last synced message or it's timestamp somewhere and sync all messages sent since this point in time.It would be useful to indicate that these messages were not synced on time and add a timestamp to show when it was originally sent.

**Describe alternatives you've considered**
To have all messages bridged, I could also let run my desktop computer 24/7 or rent a root server somewhere in a data center. But both solutions would increase the costs for running a bridge and would add much more complexity.

**Comment by a2016z** [comment]
Is the Matterbridge use case described above possible at all with the current implementation of the chat protocols?

**Comment by bugsug** [comment]
How realizable would the integration of subsequent syncing of messages be?

I could try to implement it myself, even if I never coded in Go. Could somebody point me into the right direction and say me which parts would have to be changed / added in which file?

**Comment by Helcaraxan** [comment]
Hello @bugsug,

Sorry that we didn't get to respond to your GH issue earlier. This feature is something that has been requested before (I believe) or maybe phrased in a different way or in order to solve a different problem.

In its current state the Matterbridge implementation of the various supported chat protocols is only focused on live message events and does not handle any form of history (for the protocols that support retrieving past messages).

Hence implementing this feature would be quite complex and far from straight-forward even for people familiar with Go.

Personally I would not close this feature-request but rather put it down for something that might come, hypothetically, at a yet undefined date. I'll let @42wim confirm though as he's the owner of the project and will have the definitive say on it.

**Comment by 42wim** [comment]
Yes, having some backlog support is also something I'd like to see. :+1:
